client/app/components/compose.coffee

```diff
@@ -50,21 +50,129 @@ module.exports = Compose = React.createClass
     getDefaultProps: ->
         layout: 'full'
 
+    getInitialState: ->
+        MessageUtils.createBasicMessage @props
+
+    isNew: ->
+        not @state.conversationID
 
     shouldComponentUpdate: (nextProps, nextState) ->
-        return not(_.isEqual(nextState, @state)) or
-            not (_.isEqual(nextProps, @props))
+        !!nextProps.accounts
 
+    componentWillReceiveProps: (nextProps) ->
+        if nextProps.message
```

Comment:
@aenario : that's what I thought first, and I deleted this part of code; but il fact it's about editing 'compose', in Draft Mailbox. When the route change from 'draftMessageList' to 'editDraftIntoMessageList' the component isnt destroyed bu t refresh, so the data is initialized into `getInitialState` ans updated in `shouldComponentUpdate`.